server/src/main/java/org/opensearch/search/suggest/phrase/NoisyChannelSpellChecker.java

```diff
@@ -91,11 +93,11 @@ public void reset(TokenStream stream) {
             public void nextToken() throws IOException {
                 anyTokens = true;
                 BytesRef term = fillBytesRef(termsRef);
-                if (requireUnigram && typeAttribute.type() == ShingleFilter.DEFAULT_TOKEN_TYPE) {
+                if (requireUnigram && DEFAULT_TOKEN_TYPE.equals(typeAttribute.type())) {
                     return;
                 }
                 anyUnigram = true;
-                if (posIncAttr.getPositionIncrement() == 0 && typeAttribute.type() == SynonymFilter.TYPE_SYNONYM) {
+                if (posIncAttr.getPositionIncrement() == 0 && SynonymFilter.TYPE_SYNONYM.equals(typeAttribute.type())) {
```

Comment:
(Style) You keep the class name for the second one but not the first one.  I think I'd prefer you keep the `ShingleFilter` class name and not static import the string.